### PR TITLE
Transport loading errata

### PR DESCRIPTION
**Pull request by NickAragua**
This adjust transport loading code so that a transport can load an infantry unit at the end of its movement, rather than at the beginning.

Also makes a minor efficiency improvement - when checking a path for "illegal" situations, the checks are only carried out if the path wasn't already illegal, short-circuiting through many unnecessary if blocks.

### CODE REVIEW

File: megamek/src/megamek/client/ui/swing/MovementDisplay.java [hunk 1]
```diff
@@ -2810,17 +2812,20 @@ private synchronized void updateLoadButtons() {
...
-                    // If the other unit is friendly and not the current entity
-                    // if it can tow the other unit, and if the other hasn't moved
-                    // then enable the "Tow" button.
-                    if (ce.canTow(other.getId())) {
-                        setTowEnabled(true);
-                        break;
-                    }
-                } // Check the next entity.
+            
+            //Now check all eligible hexes for towable trailers
```
Comment (Windchild292) [inline megamek/src/megamek/client/ui/swing/MovementDisplay.java:2816]:
```suggestion
            // Now check all eligible hexes for towable trailers            
```

File: megamek/src/megamek/common/MovePath.java [hunk 2]
```diff
@@ -311,15 +310,109 @@ protected MovePath addStep(final MoveStep step, boolean compile) {
...
+                                s.getElevation(), s));
+                s.setPastDanger(s.isPastDanger() || s.isDanger());
+                prevStep = s;
+            }
+        }
+        
+        if (step.useAeroAtmosphere(game, entity)
+        		&& game.getBoard().onGround()											//we're an aerospace unit on a ground map
+        		&& step.getPosition() != null  											//null
+        		&& game.getFirstEnemyEntity(step.getPosition(), entity) != null) {
```
Comment (Windchild292) [inline megamek/src/megamek/common/MovePath.java:388]: Brackets

File: megamek/src/megamek/common/MovePath.java [hunk 3]
```diff
@@ -311,15 +310,109 @@ protected MovePath addStep(final MoveStep step, boolean compile) {
...
+            MoveStep lastStep = steps.get(steps.size() - 1);
+            MoveStep prevStep = steps.get(0);
+            for (MoveStep s : steps) {
+                if (s.getMovementType(false) == EntityMovementType.MOVE_ILLEGAL) {
+                    break;
+                }
+                s.setDanger(s.isDanger() || Compute.isPilotingSkillNeeded(game, entity.getId(),
+                        prevStep.getPosition(), s.getPosition(), lastStep.getMovementType(true),
+                        prevStep.isTurning(), prevStep.isPavementStep(), prevStep.getElevation(),
+                                s.getElevation(), s));
```
Comment (Windchild292) [inline megamek/src/megamek/common/MovePath.java:379]: Odd formatting on the last line of these three... they should probably be equal

File: megamek/src/megamek/common/MovePath.java [hunk 4]
```diff
@@ -311,15 +310,109 @@ protected MovePath addStep(final MoveStep step, boolean compile) {
...
+        
+        if (shouldMechanicalJumpCauseFallDamage()) {
+            step.setDanger(true);
+        }
+                
+        // If we are using turn modes, go back through the path and mark danger for any turn
+        // that now exceeds turn mode requirement. We want to show danger on the previous step
+        // so the StepSprite will show danger. Hiding the previous step instead would make turning costs
+        // show in the turning hex for units tracking turn mode, unlike other units.
+        if (entity.usesTurnMode() && getMpUsed() > 5) {
```
Comment (Windchild292) [inline megamek/src/megamek/common/MovePath.java:354]: Brackets

File: megamek/src/megamek/common/MovePath.java [hunk 5]
```diff
@@ -311,15 +310,109 @@ protected MovePath addStep(final MoveStep step, boolean compile) {
...
+            int turnMode = getMpUsed() / 5;
+            int nStraight = 0;
+            MoveStep prevStep = steps.get(0);
+            for (MoveStep s : steps) {
+                if (s.isTurning() && nStraight < turnMode) {
```
Comment (Windchild292) [inline megamek/src/megamek/common/MovePath.java:359]: Brackets

File: megamek/src/megamek/common/MovePath.java [hunk 6]
```diff
@@ -311,15 +310,109 @@ protected MovePath addStep(final MoveStep step, boolean compile) {
...
+        if (shouldMechanicalJumpCauseFallDamage()) {
+            step.setDanger(true);
+        }
+                
+        // If we are using turn modes, go back through the path and mark danger for any turn
+        // that now exceeds turn mode requirement. We want to show danger on the previous step
+        // so the StepSprite will show danger. Hiding the previous step instead would make turning costs
+        // show in the turning hex for units tracking turn mode, unlike other units.
+        if (entity.usesTurnMode() && getMpUsed() > 5) {
+            int turnMode = getMpUsed() / 5;
```
Comment (Windchild292) [inline megamek/src/megamek/common/MovePath.java:355]: Integer division in floating point context

**Review by Windchild292 — COMMENTED** [review]
Changes look great, just a few pre-existing issues to fix (missing brackets, missing spacing, and bug with missing rounding).

File: megamek/src/megamek/common/MovePath.java (hunk 6 above)
Comment (NickAragua) [reply megamek/src/megamek/common/MovePath.java:355]: I don't know which way it rounds, and I don't want to guess, so I'm going to leave it as is.

**Comment by NickAragua** [comment]
Also got rid of some debugging code - it doesn't impact performance *too* much, but since AddStep is called a lot and we go through a lot of MovePaths, it adds up.

**Review by Windchild292 — APPROVED** [review]

**Pull request merged**